src/svg.resize.js

```diff
@@ -81,40 +84,74 @@ class ResizeHandler {
 
     const box = new Box(this.box)
 
-    if (this.events.includes('l')) {
+    if (this.eventType.includes('l')) {
       box.x = Math.min(x, this.box.x2)
       box.x2 = Math.max(x, this.box.x2)
     }
 
-    if (this.events.includes('r')) {
+    if (this.eventType.includes('r')) {
       box.x = Math.min(x2, this.box.x)
       box.x2 = Math.max(x2, this.box.x)
     }
 
-    if (this.events.includes('t')) {
+    if (this.eventType.includes('t')) {
       box.y = Math.min(y, this.box.y2)
       box.y2 = Math.max(y, this.box.y2)
     }
 
-    if (this.events.includes('b')) {
+    if (this.eventType.includes('b')) {
       box.y = Math.min(y2, this.box.y)
       box.y2 = Math.max(y2, this.box.y)
     }
 
     box.width = box.x2 - box.x
     box.height = box.y2 - box.y
 
-    if (this.el.dispatch('resize', { box: new Box(box), event: e, handler: this }).defaultPrevented) {
+    if (this.el.dispatch('resize', { box: new Box(box), angle: 0, shear: 0, eventType: this.eventType, event: e, handler: this }).defaultPrevented) {
       return
     }
 
     this.el.move(box.x, box.y).size(box.width, box.height)
   }
 
+  rotate (e) {
+    const endPoint = this.el.point(getCoordsFromEvent(e))
+    const cx = this.box.cx
+    const cy = this.box.cy
+    const dx1 = this.startPoint.x - cx
+    const dy1 = this.startPoint.y - cy
+    const dx2 = endPoint.x - cx
+    const dy2 = endPoint.y - cy
+    const c = Math.sqrt(dx1 * dx1 + dy1 * dy1) * Math.sqrt(dx2 * dx2 + dy2 * dy2)
+    if (c === 0) {
+      return
+    }
+    let angle = Math.acos((dx1 * dx2 + dy1 * dy2) / c) / Math.PI * 180
+    if (endPoint.x < this.startPoint.x) {
+      angle = -angle
+    }
+
+    this.angle = angle
+    if (this.el.dispatch('resize', { box: this.startBox, angle: angle, shear: 0, eventType: this.eventType, event: e, handler: this }).defaultPrevented) {
+      return
+    }
+
+    this.el.rotate(angle)
+  }
+  shear (e) {
+    // TODO: Add shear
+  }
   endResize (ev) {
-    this.resize(ev)
-    this.events = []
-    off(window, 'mousemove.resize touchmove.resize', this.resize)
+    // Unbind resize and end events to window
+    if (this.eventType === 'rot') { // rotate
+      off(window, 'mousemove.resize touchmove.resize', this.rotate)
+    } else if (this.eventType === 'shear') { // shear
+      off(window, 'mousemove.resize touchmove.resize', this.shear)
+    } else { // resize
+      this.resize(ev)
+      off(window, 'mousemove.resize touchmove.resize', this.resize)
+    }
```

Comment:
Alright we can do better. Just use `off(window, 'mousemove.resize touchmove.resize')` for everything!